src/Compilers/CSharp/Portable/Symbols/TypeSymbolExtensions.cs

```diff
@@ -1278,6 +1278,31 @@ private static bool HasNamespaceName(NamespaceSymbol @namespace, string namespac
             return (name.Length == length) && (string.Compare(name, 0, namespaceName, offset, length, comparison) == 0);
         }
 
+        internal static bool IsSpanType(this TypeSymbol type)
+        {
+            if ((type as NamedTypeSymbol)?.Arity != 1)
+            {
+                // must be a generic type of arity '1'
+                return false;
+            }
+
+            if (type.Name != "Span" && type.Name != "ReadOnlySpan")
+            {
+                // must be called "Span" or "ReadOnlySpan"
+                return false;
+            }
+
+            var ns = type.ContainingSymbol as NamespaceSymbol;
+            if (ns?.Name != "System")
+            {
+                // must be in "System" namespace
+                return false;
+            }
+
+            // the "System" must be in the global namespace
+            return ns.ContainingNamespace.IsGlobalNamespace;
```

Comment:
> It might be worth adding the check anyways

I agree with @OmarTawfik it is better to protect the code from some erroneous symbols. 

---
In reply to: [120762517](https://github.com/dotnet/roslyn/pull/20058#discussion_r120762517) [](ancestors = 120762517)